### DISCUSSION TITLE
Adonis Drive Permission Problem

### DISCUSSION THREAD

**Discussion by atakansn** [adonisjs/core#4706] [Help]
I can access the config settings in the @adonisjs/drive wrapper even if the visibility is private

```services: {
    fs: services.fs({
      location: app.makePath(‘storage’),
      serveFiles: true,
      routeBasePath: ‘/uploads’,
      visibility: ‘private’,
    }),
  },
```

localhost:3333/storage/uploads/avatars/1.png I can access
localhost:3333/uploads/uploads/avatars/1.png access denied error because visibility is private

How can I solve this? Or am I using it wrong?

**Comment by thetutlage** [comment]
Can you please create a reproduction of the same and share the repo?

**Reply by perogeremmer** [reply]
Hi Virk,

I guess I have the same problem. Let me tell you the short story.

I create new project and install Drive library. Once I realized I could access my storage folder like the author tell us about, and I try to access `env.ts` and it opened.

![image](https://github.com/user-attachments/assets/02f64ef9-4de8-47f1-8637-6432d53d8e16)

I'd tests several files and I thought my project could access everything because something wrong with the Drive library. 

Then try to install the new project and I still could access the `env.ts` when running with `npm run dev`. Then I decided to build the project and just run with `node bin/server.js`. I realize when it already build into the production we can't access the `env.ts` anymore.

I also think at the first time there's something wrong with your `drive.ts` config because I already change the `location: app.makePath('storage')` into `storage-new` but I could still access the storage folder through `npm run dev`.

When I try to access with private config and accessing through `/uploads` path yes the return is `Access Denied`.

![image](https://github.com/user-attachments/assets/e47614d1-eca0-4336-a783-3e02f77f23b3)

**So I think the problem is the author running the local version like me and couldn't understand why we can access every single of files in our project and think the library is wrong or having any bugs.**

--- 

My suggestion, I hope you can limit the access so we can't show every file in browser using `GET`. Overall, I really love Adonis ❤️

Thank you for making such a great framework for us 🙂

**Comment by perogeremmer** [comment]
Hi Author, I guess the problem is because you're running the local server.

Try to build the project, run the project using node and re-access the same path, I guess you will see now you can't access `/storage`.

**Comment by thetutlage** [comment]
Please share a sample project that I can use for reproduction
